src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -61,6 +72,11 @@ public string Arguments
             set => _arguments = value;
         }
 
+        public IReadOnlyCollection<string> ArgumentList
+        {
+            get => _argumentList;
```

Comment:
How does someone add arguments?

Reply:
Oh, I see, we require that the arguments be passed in to the PSI constructor.  That seems to go against the rest of the PSI API, where you can create a PSI and then add to collections, set properties, etc.  Is there a reason this one differs? (I'll go back and read the design meeting notes.)

Reply:
I just skimmed through that portion of the design meeting video.  I could certainly have missed it, but I didn't see a mention of the constructor, and it seems like this design of not having a setter was predicated on the property returning a mutable collection type, and then later it was modified to return a read-only one.  @danmosemsft, what's the design here?  Am I just misunderstanding?

Reply:
We have a constructor for the flattened arguments case - (`ProcessStartInfo(string fileName, string arguments)`) so it's not inconsistent to have one for the array version. But there are two problems:

1. I don't remember the constructor being approved.

2. This is a potentially source-breaking addition since the ctor call becomes ambiguous now if `null` is passed.

Reply:
> This is a potentially source-breaking addition since the ctor call becomes ambiguous now if null is passed.

Yes, and whereas in some cases we've allowed that in the past, in particular where it would have resulted in an exception to pass null as the argument, here it's perfectly valid to pass null as `string arguments`, and I would not be surprised if it were common to do so.

Reply:
I asked about adding the constructor from Immo. I have added @stephentoub you on the thread and @AtsushiKan you are already on it. It didn`t make sense of having read only property and no was to initialize it.

Reply:
About the ```null``` being passed, that's an implementation detail, we can change it as we like.